MvvmCross-AndroidSupport/MvvmCross.Droid.Support.V7.AppCompat/MvxAppCompatViewPresenter.cs

```diff
@@ -0,0 +1,280 @@
+           MvxDialogAttribute attribute,
+           MvxViewModelRequest request)
+        {
+            var fragmentName = FragmentJavaName(attribute.ViewType);
+            Dialog = (DialogFragment)CreateFragment(fragmentName);
+            //TODO: Find a better way to set the ViewModel at the Fragment
+            ((IMvxFragmentView)Dialog).ViewModel = ((MvxViewModelInstanceRequest)request).ViewModelInstance;
+            Dialog.Cancelable = attribute.Cancelable;
+            Dialog.Show(CurrentFragmentManager, fragmentName);
+        }
+
+        protected override void ShowActivity(Type view, MvxActivityAttribute attribute, MvxViewModelRequest request)
+        {
+            var intent = CreateIntentForRequest(request);
+            if(attribute.Extras != null)
+                intent.PutExtras(attribute.Extras);
+
+            var activity = CurrentActivity;
+            if (activity == null)
+            {
+                MvxTrace.Warning("Cannot Resolve current top activity");
+                return;
+            }
+
+            if (attribute.SharedElements != null)
+            {
+                IList<Pair> sharedElements = new List<Pair>();
+                foreach (var item in attribute.SharedElements)
+                {
+                    intent.PutExtra(item.Key, ViewCompat.GetTransitionName(item.Value));
+                    sharedElements.Add(Pair.Create(item.Value, item.Key));
+                }
+                ActivityOptionsCompat options = ActivityOptionsCompat.MakeSceneTransitionAnimation(CurrentActivity, sharedElements.ToArray());
+                activity.StartActivity(intent, options.ToBundle());
+            }
+            else
+                activity.StartActivity(intent);
+        }
+
+        protected override IMvxFragmentView CreateFragment(string fragmentName)
+        {
+            var fragment = Fragment.Instantiate(CurrentActivity, fragmentName);
+            return (IMvxFragmentView)fragment;
+        }
+
+        protected override async Task ShowHostActivity(MvxFragmentAttribute attribute)
+        {
+            var currentHostViewModelType = GetCurrentActivityViewModelType();
+            if (attribute.ActivityHostViewModelType != currentHostViewModelType)
+            {
+                var viewType = ViewsContainer.GetViewType(attribute.ActivityHostViewModelType);
+                if (!viewType.IsSubclassOf(typeof(FragmentActivity)))
+                    throw new MvxException("The host activity doesnt inherit FragmentActivity");
+
+                var hostViewModelRequest = MvxViewModelRequest.GetDefaultRequest(attribute.ActivityHostViewModelType);
+                Show(hostViewModelRequest);
+
+                int tries = 10;
+                while ((GetCurrentActivityViewModelType() != attribute.ActivityHostViewModelType) && (tries > 0))
+                {
+                    await Task.Delay(1);
+                    tries--;
+                }
```

Comment:
@Nickolas- I thought about that one too, but it means we need to add that, and it can also conflict with other `Application` classes like James Montemagno CurrentActivity Plugin.